docs/DeveloperGuide.md

```diff
@@ -1,34 +1,938 @@
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;4.5.2. [Preview Upcoming Dues Feature](#452-preview-upcoming-dues-feature)<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;4.5.3. [Exclusion Feature](#453-exclusion-feature)<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;4.5.4. [Reschedule Chapter Feature](#454-reschedule-chapter-feature)<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;4.5.5. [View Revision History Feature](#455-view-revision-history-feature)<br>
+5. [<b>Appendix: Requirements</b>](#5-appendix-requirements)<br>
+5.1. [Product Scope](#51-product-scope)<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;5.1.1. [Target User Profile](#511-target-user-profile)<br>
+&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;5.1.2. [Value Proposition](#512-value-proposition)<br>
+5.2. [User Stories](#52-user-stories)<br>
+5.3. [Use Cases](#53-use-cases)<br>
+5.4. [Non-Functional Requirements](#54-non-functional-requirements)<br>
+5.5. [Glossary](#55-glossary)<br>
+6. [<b>Appendix: Instructions for Manual Testing</b>](#6-appendix-instructions-for-manual-testing)<br>
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+--------------------------------------------------------------------------------------------------------------------
 
+## 1. Introduction
 
-## Product scope
-### Target user profile
+### 1.1. Overview
 
-{Describe the target user profile}
+### 1.2. Purpose
 
-### Value proposition
+### 1.3. Scope
+This documentation describes the software architecture and software design decisions for the implementation of Kaji. The intended audience of this document is the developers, designers, and software testers of Kaji.
 
-{Describe the value proposition: what problem does it solve?}
+--------------------------------------------------------------------------------------------------------------------
 
-## User Stories
+## 2. Setting Up
+
+### 2.1. Prerequisites
+* JDK 11
+* IntelliJ IDEA
+
+### 2.2. Setting up the project in your computer
+1. Fork this repository, and clone the fork into your computer.
+2. Open IntelliJ (if you are not in the welcome screen, click `File` → `Close Project` to close the existing project dialog first).
+3. Set up the correct JDK version for Gradle.
+    1. Click `Configure` → `Project Defaults` → `Project Structure`.
+    2. Click `New…` and set it to the directory of the JDK.
+4. Click `Import Project` (or `Open or Import` in newer version of Intellij).
+5. Locate the `build.gradle` file (not the root folder as you would do in a normal importing) and select it. Click `OK`.
+If asked, choose to `Open as Project` (not `Open as File`).
+7. Click `OK` to accept the default settings
+
+--------------------------------------------------------------------------------------------------------------------
+
+## 3. Design (Architecture)
+(Zeyu)
+
+![Architecture Diagram of Design Component](UML/Architecture.png)
```

Comment:
I think it might be better if your diagrams were standardized as to how they were added into the document. Some of them use the HTML notation, aligned and include caption, the rest just follows a simple markdown format. That is not so much of an issue, but maybe the alignment and the use of caption can be standardized?

Reply:
Maybe can consider labelling the circled component and state that that is a human-editable output file? It wasn't immediately clear for me, and I had to think of what would come out from Storage. It helped that almost all of us had similar architecture for our projects, may not be for a third-party developer.
![Screenshot_1](https://user-images.githubusercontent.com/54029207/97528616-2728ed00-19e9-11eb-8d99-a3ef13d28e11.png)